### ISSUE TITLE
How to do face effects

### ISSUE THREAD

**Issue by tjasmin111** [wysaid/android-gpuimage-plus#522]
Is it possible to use it to do facial effects such as making nose smaller or eyes bigger? How can we do that with good quality like filters we see on Instagram or Snapchat?

Any tips or samples would be appreciated.

**Comment by wysaid** [comment]
Sure, checkout the old branch here: https://github.com/wysaid/android-gpuimage-plus/tree/face_features

I did not previously find a suitable open-source library for face landmark recognition, and the functionality of the one I have chosen is not as accurate either.

**Comment by wysaid** [comment]
> Sure, checkout the old branch here: https://github.com/wysaid/android-gpuimage-plus/tree/face_features
> 
> I did not previously find a suitable open-source library for face landmark recognition, and the functionality of the one I have chosen is not as accurate either.

So it's deprecated for now.

**Comment by tjasmin111** [comment]
Why don't you update that Branch? I think an automatic filter would be more desirable than click based

**Comment by wysaid** [comment]
I didn't find a suitable open-source library for face landmark detection at that time. However, if you know any, could you please recommend a few to me?

**Comment by tjasmin111** [comment]
Yes, google ML kit looks good. I played with it a bit. It has both face landmark detection and face mesh. I just don't know how to use them for facial effects and warping.

**Comment by wysaid** [comment]
Looks good, I'll give it a try, but it won't be so fast. You can also try providing a PR for this feature.

**Comment by tjasmin111** [comment]
I don't have much to contribute but sure. I'll stay tuned. This a lovely feature and let's do it!

**Comment by WoYang** [comment]
try MediaPie
